openlibrary/plugins/openlibrary/js/initTaggingSearchBar.js

```diff
@@ -0,0 +1,201 @@
+import { debounce } from './nonjquery_utils';
+
+const subjectTypeColors = {
+    subject: '#0067D5',
+    person: '#D100D5',
+    place: '#D50033',
+    time: '#D5A600'
+}
+
+function newSubjectRowHtml(subjectName, subjectType = null, isSelected = false) {
+    const exists = subjectType != null;
+    const buttonStyle = 'font-size: 13px; color: black; font-weight: 700; cursor: pointer; width: 40px;';
+    const nameStyle = 'font-size: 13px; color: black; font-weight: 400';
+    const tagStyle = 'font-size: 11px; font-weight: 400; border-radius: 8px; border-top: 1px solid #CCC; border-bottom: 1px solid #CCC; padding: 2px 6px; color: #FFF; margin-left: auto';
+    const divStyle = 'display: flex; align-items: center; font-size: 16px; border-bottom: 1px solid lightgray; padding: 9px 10px; width: 100%;';
+
+    const div = document.createElement('div');
+    div.style = divStyle;
+    if (exists) {
+        div.style.height = '33px';
+    }
+
+    if (exists) {
+        const buttonIcon = isSelected ? '-' : '+';
+        const button = document.createElement('div');
+        button.innerText = buttonIcon;
+        button.style = buttonStyle;
+        button.addEventListener('click', () => handleSelectSubject(subjectName, subjectType, button));
```

Comment:
Thinking on whether this might be cleaner with Vue; eg. Let me work on describing the Vue flow.

```suggestion
    <div class="FOO_button" @click="handleSelectSubect(name, type)>{{isSelected ? '-' : '+'}}</div>
```